examples/src/bin/unbuffered-async-client.rs

```diff
@@ -0,0 +1,245 @@
+use std::error::Error;
+use std::sync::Arc;
+
+use helpers::KB;
+use rustls::client::{ClientConnectionData, UnbufferedClientConnection};
+use rustls::unbuffered::{
+    AppDataRecord, ConnectionState, EncodeError, EncryptError, MayEncryptAppData, UnbufferedStatus,
+};
+#[allow(unused_imports)]
+use rustls::version::{TLS12, TLS13};
+use rustls::{ClientConfig, RootCertStore};
+use rustls_examples as helpers;
+
+#[cfg(not(feature = "async-std"))]
+use tokio::{
+    io::{AsyncReadExt, AsyncWriteExt},
+    net::TcpStream,
+};
+
+// async-std
+// to switch the runtime modify `Cargo.toml` and the `main` attribute below
+#[cfg(feature = "async-std")]
+use async_std::{
+    io::{ReadExt, WriteExt},
+    net::TcpStream,
+};
+
+const SERVER_NAME: &str = "example.com";
+const PORT: u16 = 443;
+
+const INCOMING_TLS_BUFSIZE: usize = 16 * KB;
+const OUTGOING_TLS_INITIAL_BUFSIZE: usize = KB;
+
+const MAX_ITERATIONS: usize = 20;
+
+#[cfg_attr(not(feature = "async-std"), tokio::main(flavor = "current_thread"))]
+#[cfg_attr(feature = "async-std", async_std::main)]
+async fn main() -> Result<(), Box<dyn Error>> {
+    let mut root_store = RootCertStore::empty();
+    root_store.extend(
+        webpki_roots::TLS_SERVER_ROOTS
+            .iter()
+            .cloned(),
+    );
```

Comment:
I think this can be simplified to:
```rust
let root_store = RootCertStore {
    roots: webpki_roots::TLS_SERVER_ROOTS.into(),
};
```

Ditto for `unbuffered-client.rs` and `unbuffered-server.rs` (sorry I missed that in the prev review).